enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/interpreted/RegisteredPipeBuilder.scala

```diff
@@ -141,7 +155,8 @@ class RegisteredPipeBuilder(fallback: PipeBuilder,
 
     plan match {
       case Apply(_,_) => ApplyRegisterPipe(lhs, rhs)(id)
-      case _ => fallback.build(plan, lhs, rhs)
+      case CartesianProduct(_, _) => fallback.build(plan, lhs, rhs)
```

Comment:
CartesianProductPipe does funky stuff with execution contexts. Are you sure this should be handled by the fallback and not just fail?